pkg/registry/fetch.go

```diff
@@ -0,0 +1,78 @@
+package registry
+
+import (
+	"fmt"
+	helmopv1 "github.com/fluxcd/helm-operator/pkg/apis/helm.fluxcd.io/v1"
+	"github.com/go-logr/logr"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/cli"
+	"log"
+	"os"
+	"path"
+)
+
+func Fetch(helmReleaseSpec helmopv1.HelmReleaseSpec, location string, logr logr.Logger) (string, error) {
```

Comment:
```suggestion
func Fetch(logr logr.Logger, helmReleaseSpec helmopv1.HelmReleaseSpec, location string) (string, error) {
```
More idiomatic to pass supporting variables like config, context or logger as the first set of args.